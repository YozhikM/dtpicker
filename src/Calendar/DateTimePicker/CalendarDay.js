/* @flow */

import React from 'react';
import { getDate } from 'date-fns';

type Props = {|
  date: Date,
  onClick?: (date: Date) => void,
  style?: any,
|};

export default class CalendarDay extends React.Component<Props, void> {
  onClick = () => {
    const { onClick, date } = this.props;
    if (onClick) {
      onClick(date);
    }
  };

  render() {
    const { date, style } = this.props;
    const dayOfMonth = getDate(date);

    return (
      <td style={style} onClick={this.onClick}>
        {dayOfMonth}
      </td>
    );
  }
}
