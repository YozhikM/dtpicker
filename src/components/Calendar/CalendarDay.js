/* @flow */

import React from 'react';
import { getDate } from 'date-fns';

type Props = {|
  date: Date,
  onClick?: (date: Date) => void,
  style: any
|};

type State = void;

class CalendarDay extends React.Component<Props, State> {
  onClick = () => {
    const { onClick, date } = this.props;
    if (onClick) {
      onClick(date);
    }
  };

  render() {
    const { date, style } = this.props;
    let dayOfMonth = getDate(date);
    return <td style={style} onClick={this.onClick}>{dayOfMonth}</td>;
  }
}

export default CalendarDay;
