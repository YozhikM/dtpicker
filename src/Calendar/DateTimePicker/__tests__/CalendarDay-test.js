/* @flow */

import React from 'react';
import CalendarDay from '../CalendarDay';

describe('CalendarDay', () => {
  describe('render()', () => {
    it('basic', () => {
      expect(shallow(<CalendarDay date={new Date(2009, 0, 9)} />)).toMatchSnapshot();
    });
  });

  describe('user interactions', () => {
    it('onClick()', () => {
      const onClick = jest.fn();
      const date = new Date(2015, 4, 5);
      const wrapper = shallow(<CalendarDay date={date} onClick={onClick} />);
      wrapper.find('td').simulate('click');

      expect(onClick).toBeCalledWith(date);
    });
  });
});
