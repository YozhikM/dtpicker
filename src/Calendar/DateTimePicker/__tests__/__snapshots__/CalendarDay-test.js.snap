// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDay render() basic 1`] = `
<td
  onClick={[Function]}
>
  9
</td>
`;
